### ISSUE TITLE
Fix block solidness

### ISSUE THREAD

**Issue by Nightenom** [ldtteam/Structurize#585]
- no leaves, or special check is they are permanent, or tag for ppl to manually adjust
- iterate block registry and collect everything that returns true from block#canSurvive(null, null, null) and doesnt NPE

pseudocode:
```
bool canFloatInAir(block)
{
 if is in tag -> false
 if is leaves -> leaves.permenent
 else -> filtered_registry.contains(block)
}
```

**Comment by Nightenom** [comment]
Closes #583

**Comment by Raycoms** [comment]
Done
